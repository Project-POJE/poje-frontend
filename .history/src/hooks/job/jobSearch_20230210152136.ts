export function useJobSearchPrefetch(keyword: string) {
  //Todo 검색결과 프리패치 함수
}

export function useJobSearch(keyword: string) {
  //Todo 검색결과 가져오는 함수
}
