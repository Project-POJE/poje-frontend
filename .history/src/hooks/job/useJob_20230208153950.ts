export default function useJob() {
  return <div></div>;
}
