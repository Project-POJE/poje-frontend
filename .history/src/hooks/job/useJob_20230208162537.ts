import { useQuery } from '@tanstack/react-query';
interface UseJobType {
  job: string;
}
export default function useJob({ job }: UseJobType) {
  const {} = useQuery();
  return 123;
}
