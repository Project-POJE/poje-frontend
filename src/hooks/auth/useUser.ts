import { accessTokenState } from '@src/store/auth/auth';
import { useRecoilValue, useRecoilState, useResetRecoilState } from 'recoil';

export default function useUser() {
  const [user, setUser] = useRecoilState(accessTokenState);
  const resetUser = useResetRecoilState(accessTokenState);
  const clearUser = () => {
    resetUser();
  };

  return { user, clearUser, setUser };
}
