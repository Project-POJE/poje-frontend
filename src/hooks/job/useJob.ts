export default function useJob() {
  return 123;
}
