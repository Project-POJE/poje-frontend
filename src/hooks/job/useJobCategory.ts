import service from '@src/service';
import { useQuery } from '@tanstack/react-query';

export default function useJobCategory() {
  const {
    data: category = [],
    isLoading,
    isSuccess,
  } = useQuery(
    ['category'],
    () => {
      return service.job.getJobCategory();
    },
    {
      onError: (e: unknown) => {
        console.log('에러에요에어에어에어에어에어에어에어');
      },
    }
  );
  return { category, isLoading, isSuccess };
}
