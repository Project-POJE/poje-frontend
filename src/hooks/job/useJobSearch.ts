import { useCallback } from 'react';
export default function useJobSearch() {
  const prefetchJobSearch = useCallback((keyword: string) => {
    //Todo 검색결과 프리패치 함수
    console.log('프리패칭 수행중 ...');
  }, []);

  const getJobListByKeyword = useCallback((keyword: string) => {
    //Todo 검색결과 반환하기
  }, []);
  return { prefetchJobSearch, getJobListByKeyword };
}
