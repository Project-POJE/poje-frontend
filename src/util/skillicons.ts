import { SkillItemType } from '@src/service/types/portfolio';

export type SkillIconSetType = {
  name: string;
  path: string;
};

export type SkillIconType = 'frontend' | 'backend';

export type IconListType = Record<SkillIconType, SkillItemType[]>;

export type IconListKeyType = keyof IconListType;

type SkillTypeAndNameType = {
  type: IconListKeyType;
  name: string;
  path: string;
};

export function extractSkillIconFromFolder() {
  //skill_icon 하위 폴더들의 이미지들을 가져옴
  const imgContext = require.context(
    '/public/public_assets/skill_icon',
    true,
    /.png$|.ipg$/
  );

  const icons: IconListType = {
    frontend: [],
    backend: [],
  };

  //./backend/node.png 이런애들이 Keys()에 있음
  imgContext.keys().forEach((k) => {
    const extractedSkill = extractSkillTypeAndName(k);
    console.log(k);
    if (extractedSkill) {
      const name = extractedSkill.name;
      const path = extractedSkill.path;
      icons[extractedSkill.type].push({ name, path });
    }
  });
  console.log(icons);

  return icons;
}

// 동적으로 키에 배열을 작성하고 난 후 그 키를 동적으로 가져오는 것은 불가능
function extractSkillTypeAndName(path: string): SkillTypeAndNameType | null {
  // 직업식별자에 포함된 이미지의 식별자를 가져오기
  let skillType, skillName, skillPath;
  let result = path.replace('./', ''); //추출된 스킬 front/css.png;
  const splitPathList = result.split('/');

  if (splitPathList.length === 1) return null;
  skillType = splitPathList[0] as IconListKeyType;
  skillName = splitPathList[1].split('/')[0].split('.')[0];
  skillPath = splitPathList[1].split('/')[0];
  return { type: skillType, name: skillName, path: skillPath };
}
