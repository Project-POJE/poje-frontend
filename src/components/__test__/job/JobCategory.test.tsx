import JobCategory from '@src/components/home/JobPage/JobCategory';
import useUser from '@src/hooks/auth/useUser';
import useJobCategory from '@src/hooks/job/useJobCategory';
import { mockJobCategory } from '@src/test_util/mock/data/job';
import { withAllContext, withMemoryRouter } from '@src/test_util/util';
import { render } from '@testing-library/react';
import { Route } from 'react-router-dom';

jest.mock('@src/hooks/auth/useUser');
jest.mock('@src/hooks/job/useJobCategory');

const mockUseUser: jest.Mock = useUser as jest.MockedFunction<any>;
const mockUseJobCategory: jest.Mock =
  useJobCategory as jest.MockedFunction<any>;

describe('JobCategoty Component', () => {
  beforeEach(() => {
    mockUseUser.mockReset();
    mockUseJobCategory.mockReset();
  });

  it('로그인 했을 때 화면', () => {
    mockUseUser.mockImplementation(() => ({ user: true }));
    mockUseJobCategory.mockImplementation(() => ({
      category: mockJobCategory,
    }));

    const { asFragment } = render(
      withAllContext(
        withMemoryRouter(<Route path='/' element={<JobCategory />} />, {
          initialEntries: ['/'],
        })
      )
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('로그인 안 했을 때 화면', () => {
    mockUseUser.mockImplementation(() => ({ user: false }));
    mockUseJobCategory.mockImplementation(() => ({
      category: mockJobCategory,
    }));

    const { asFragment } = render(
      withAllContext(
        withMemoryRouter(<Route path='/' element={<JobCategory />} />, {
          initialEntries: ['/'],
        })
      )
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
