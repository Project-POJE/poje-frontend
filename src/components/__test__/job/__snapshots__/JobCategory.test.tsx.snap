// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCategoty Component 로그인 안 했을 때 화면 1`] = `
<DocumentFragment>
  <ul
    class="sc-bcXHqe fUoPjS"
  >
    <li
      class="sc-gswNZR froitf"
    >
      <a
        class=""
        href="/job/개발자/1"
      >
        개발자
      </a>
    </li>
    <li
      class="sc-gswNZR froitf"
    >
      <a
        class=""
        href="/job/디자이너/1"
      >
        디자이너
      </a>
    </li>
    <li
      class="sc-gswNZR froitf"
    >
      <a
        class=""
        href="/job/유승윤/1"
      >
        유승윤
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`JobCategoty Component 로그인 했을 때 화면 1`] = `
<DocumentFragment>
  <ul
    class="sc-bcXHqe fUoPjS"
  >
    <li
      class="sc-gswNZR froitf"
    >
      <a
        class=""
        href="/job/개발자/1"
      >
        개발자
      </a>
    </li>
    <li
      class="sc-gswNZR froitf"
    >
      <a
        class=""
        href="/job/디자이너/1"
      >
        디자이너
      </a>
    </li>
    <li
      class="sc-gswNZR froitf"
    >
      <a
        class=""
        href="/job/유승윤/1"
      >
        유승윤
      </a>
    </li>
    <li
      class="sc-gswNZR froitf"
    >
      <a
        class=""
        href="/job/like/1"
        style="color: black;"
      >
        좋아요 누른 포트폴리오
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
